### PR TITLE
Add support for git-config

### LINKED ISSUE #54: support git config options
support custom git config options

```
git config --add core.gitproxy '"proxy-command" 
```

as a yaml slice parameter:

```
git_config:
  - core.gitproxy proxy-command
  - user.email john@doe.org
```

### PR DESCRIPTION
**Pull request by typeless**
Fixes #54 

Not tested yet.
Will test it later on.

### CODE REVIEW

**Comment by techknowlogick** [comment]
Hi @typeless, this repo has moved to https://github.com/drone/drone-git, you may want to move this PR over there.

**Comment by typeless** [comment]
@techknowlogick got it

File: main.go [hunk 1]
```diff
@@ -151,5 +157,13 @@ func run(c *cli.Context) error {
 		},
 	}
 
+	for _, s := range c.StringSlice("git-configs") {
```
Comment (tboerger) [inline main.go:160]: IMHO you don't even need to split these strings, just keep the slice of strings and iterate through that with `git config --global STRING`

**Comment by tboerger** [comment]
The official default clone plugin is at https://github.com/drone/drone-git, but I would still accept that here as well.

File: main.go (hunk 1 above)
Comment (typeless) [reply main.go:160]: I tried at my Linux box, it didn't work.
image

Comment (tboerger) [reply main.go:160]: You are right, it must be supplied as to attributes.

**Comment by typeless** [comment]
Moved to https://github.com/drone-plugins/drone-git/pull/87.

**Pull request closed without merging**